Lib/idlelib/grep.py

```diff
@@ -45,36 +73,54 @@ def open(self, text, searchphrase, io=None):
         self.globvar.set(os.path.join(dir, "*" + tail))
 
     def create_entries(self):
+        "Create base entry widgets and add widget for search path."
         SearchDialogBase.create_entries(self)
         self.globent = self.make_entry("In files:", self.globvar)[0]
 
     def create_other_buttons(self):
+        "Add check button to recurse down subdirectories."
         btn = Checkbutton(
                 self.make_frame()[0], variable=self.recvar,
                 text="Recurse down subdirectories")
         btn.pack(side="top", fill="both")
 
     def create_command_buttons(self):
+        "Create base command buttons and add button for search."
         SearchDialogBase.create_command_buttons(self)
         self.make_button("Search Files", self.default_command, 1)
 
     def default_command(self, event=None):
+        """Grep for search pattern in file path.
+
+        The default command is bound to <Return>.
+
+        If entry values are populated, set OutputWindow as stdout
+        and perform search.
+        """
         prog = self.engine.getprog()
         if not prog:
-            return
+            return None
         path = self.globvar.get()
         if not path:
             self.top.bell()
-            return
+            return None
         from idlelib.outwin import OutputWindow  # leave here!
         save = sys.stdout
         try:
             sys.stdout = OutputWindow(self.flist)
             self.grep_it(prog, path)
         finally:
             sys.stdout = save
+        return None
```

Comment:
" If any return statement returns an expression, any return statements where no value is returned should explicitly state this as return None "  So if no return statement return an expression, no explicit return is needed at the end.  In other words, this function was within the rules and should be left as it was.